### ISSUE TITLE
Responses to gossip requests don't originate from the port the request was received on (9634)

### ISSUE THREAD

**Issue by mivok** [habitat-sh/habitat#1146]
Symptoms: If you try to run habitat on two machines where UDP port 9634 is allowed inbound, but no other UDP ports, then habitat peers can't communicate with each other.

Habitat version: hab 0.8.0/20160729175817

To reproduce:
- Start two ec2 instances running ubuntu 16.04, in a vpc (the details here probably don't matter, but I include them just in case). The security group on the instance should allow port 9634 inbound from the vpc network, and all traffic outbound, but no other UDP ports. TCP ports don't matter, although you should probably allow port 22 at least.
- Install habitat and make the hab user/group
- Run `sudo hab start core/redis` on the first instance, and note down the IP reported by habitat (I'll use 172.16.0.10 as an example here)
- In another window, run `sudo tcpdump -n udp and not port 123` on the first instance.
- On the second instance, run `sudo hab start core/redis --peer 172.16.0.10`
- The connection will fail
- The tcpdump output will show something similar to the following:
  
  ```
  mharrison@ip-172-16-0-10:~$ sudo tcpdump -n udp and not port 123
  tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
  listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
  22:16:20.580419 IP 172.16.0.11.47355 > 172.16.0.10.9634: UDP, length 20
  22:16:20.580486 IP 172.16.0.10.57737 > 172.16.0.11.47355: UDP, length 20
  22:16:21.577795 IP 172.16.0.11.47355 > 172.16.0.10.9634: UDP, length 20
  22:16:21.577935 IP 172.16.0.10.55924 > 172.16.0.11.47355: UDP, length 20
  22:16:21.579563 IP 172.16.0.10.57737 > 172.16.0.11.47355: UDP, length 20
  22:16:21.579576 IP 172.16.0.10.57737 > 172.16.0.11.47355: UDP, length 20
  ```

Note that the request comes from port 47355 to port 9634, but the response comes from port 57737 to port 47355. The expectation is that the response would originate from port 9634, which would allow the packet to pass stateful firewalls.

**Comment by reset** [comment]
@mivok thanks for doing some digging here! We've hit this ourselves and we'll get this fixed soon

**Comment by metadave** [comment]
[This](https://github.com/maidsafe-archive/rust-utp/commit/1f2b8e834479066318aaf79eeebea24177685d12) connection method in µTP may be worth looking in to.

**Comment by smurawski** [comment]
This should be resolved in the recent supervisors (0.13.0 and newer iirc).

**Comment by adamhjk** [comment]
It is resolved. We now use the address that the other end of the socket has as the address we use to route messages to that member.
